aten/src/ATen/native/cuda/MaximumMinimum.cu

```diff
@@ -0,0 +1,94 @@
+#include <ATen/AccumulateType.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/BinaryOps.h>
+#include <ATen/native/DispatchStub.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cuda/Loops.cuh>
+
+// NOTE: CUDA on Windows requires that the enclosing function
+// of a __device__ lambda not have internal linkage.
+
+namespace at { namespace native {
+
+void maximum_kernel_cuda(TensorIterator& iter) {
+  if (isIntegralType(iter.dtype(), /*includeBool=*/ true)) {
+    AT_DISPATCH_INTEGRAL_TYPES_AND(at::ScalarType::Bool, iter.dtype(), "maximum_cuda", [&] {
+      gpu_kernel(iter, [] GPU_LAMBDA (scalar_t a, scalar_t b) -> scalar_t {
+        return a >= b ? a : b;
+      });
+    });
+  } else if (isFloatingType(iter.dtype())) {
+    AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.input_dtype(), "maximum_cuda", [&]() {
+      gpu_kernel(iter, [] GPU_LAMBDA (scalar_t a, scalar_t b) -> scalar_t {
+        // isnan(half) breaks the Windows build. We explicitly cast half to float.
+        using acc_type = typename AccumulateType<scalar_t, /*is_cuda=*/true>::type;
```

Comment:
Why use the accumulate type here?

Reply:
I was referring the following code from `MaxMinElementwiseKernel.cu` to prevent `isnan` breaks for `half` types on the Windows platform.
https://github.com/pytorch/pytorch/blob/e8f4b04d9ae1880fd8d09dde40639e86a0d2c585/aten/src/ATen/native/cuda/MaxMinElementwiseKernel.cu#L26-L32

And thanks to the kind suggestions on using `a != a` directly, the `accumulate type` is no longer needed.